### ISSUE TITLE
Node name pattern matching broken on 'node set'

### ISSUE THREAD

**Issue by severgun** [warewulf/warewulf#964]
### Steps to reproduce

```console
wwctl node add node[00-09]
wwctl node list node[00-09]

NODE NAME              PROFILES                   NETWORK
================================================================================
node02                 default                    default
node03                 default                    default
node04                 default                    default
node06                 default                    default
node00                 default                    default
node01                 default                    default
node05                 default                    default
node07                 default                    default
node08                 default                    default
node09                 default                    default
```
Unsorted but OK.
```console
wwctl profile add compute
wwctl node set --profile compute node[00-09] # 0 nodes
wwctl node set --profile compute node0[0-9] # 10 nodes
```

other example:
I tried nodes with FQDN names:

```console
wwctl node add n[00-19].example.com
wwctl node list n[00-19].example.com
NODE NAME              PROFILES                   NETWORK
================================================================================
n02.example.com        default                    default
n11.example.com        default                    default
n14.example.com        default                    default
n05.example.com        default                    default
n07.example.com        default                    default
n08.example.com        default                    default
n12.example.com        default                    default
...
```
Works OK

```console
wwctl node set --profile compute n[00-19].example.com # 0 nodes
wwctl node set --profile compute n[00-19].example.* # 0 nodes
wwctl node set --profile compute n[00-19].* # 30 nodes
wwctl node set --profile compute n[01-19].* # 30 nodes
wwctl node set --profile compute n[02-19].* # 0 nodes
```

### Information on your system

wwctl version:   4.4.1-1.git_d6f6fed
rpc version: apiPrefix:"rc1" apiVersion:"1" warewulfVersion:"4.4.1-1.git_d6f6fed"

NAME="Rocky Linux"
VERSION="8.8 (Green Obsidian)"
ID="rocky"
ID_LIKE="rhel centos fedora"
VERSION_ID="8.8"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Rocky Linux 8.8 (Green Obsidian)"
ANSI_COLOR="0;32"
LOGO="fedora-logo-icon"
CPE_NAME="cpe:/o:rocky:rocky:8:GA"
HOME_URL="https://rockylinux.org/"
BUG_REPORT_URL="https://bugs.rockylinux.org/"
SUPPORT_END="2029-05-31"
ROCKY_SUPPORT_PRODUCT="Rocky-Linux-8"
ROCKY_SUPPORT_PRODUCT_VERSION="8.8"
REDHAT_SUPPORT_PRODUCT="Rocky Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="8.8"

### General information

- [X] I have run `wwctl version` and reported the contents of `/etc/os-release`
- [X] I have searched the issues of this repo and believe this is not a duplicate
- [X] I have captured and reported relevant error messages and logs

**Comment by mslacken** [comment]
Thanks for reporting. 
The underlying problem is that only for the `node add` command the `[]` are expanded and also only one pair of brackets.
For `node list` there is no expansion, but the argument is simply treated as regexp.
So yes this needs to be fixed in the `4.6.x` release.

**Comment by severgun** [comment]
```console 
wwctl node set --profile compute_ip node[00-19].example.com # 0 nodes
wwctl node list node[00-09].example.com # 20 nodes
```
for `node set` it is regex. For example `wwctl node set --profile compute_ip node[0-9]{2}.example.com` return 20 nodes as expected.
but for list it is weird
```console
wwctl node list node[0-9]{2}.example.com
NODE NAME              PROFILES                   NETWORK
================================================================================
node00.example.com     default                    default
node11.example.com     default                    default
```

```console
wwctl node list node[00-19].example.com
NODE NAME              PROFILES                   NETWORK
================================================================================
node02.example.com     default                    default
node03.example.com     default                    default
node08.example.com     default                    default
node09.example.com     default                    default
node00.example.com     default                    default
node04.example.com     default                    default
node05.example.com     default                    default
node10.example.com     default                    default
node15.example.com     default                    default
node18.example.com     default                    default
node01.example.com     default                    default
node07.example.com     default                    default
node11.example.com     default                    default
node14.example.com     default                    default
node19.example.com     default                    default
node06.example.com     default                    default
node12.example.com     default                    default
node13.example.com     default                    default
node16.example.com     default                    default
node17.example.com     default                    default
```
